curves/bn254/icicle.go

```diff
@@ -1,105 +1,135 @@
 package bn254
 
 import (
-	"fmt"
-	"math"
-	"unsafe"
-
-	"github.com/consensys/gnark-crypto/ecc/bn254"
-	"github.com/consensys/gnark-crypto/ecc/bn254/fp"
-	goicicle "github.com/ingonyama-zk/icicle/goicicle"
-	icicle "github.com/ingonyama-zk/icicle/goicicle/curves/bn254"
+	"errors"
+
+	bn254 "github.com/consensys/gnark-crypto/ecc/bn254"
+	"github.com/consensys/gnark-crypto/ecc/bn254/fr"
+	"github.com/ingonyama-zk/icicle/wrappers/golang/core"
+	cr "github.com/ingonyama-zk/icicle/wrappers/golang/cuda_runtime"
+	icicle_bn254 "github.com/ingonyama-zk/icicle/wrappers/golang/curves/bn254"
 )
 
-type OnDeviceData struct {
-	P    unsafe.Pointer
-	Size int
+func MsmOnDevice(gnarkPoints []bn254.G1Affine, gnarkScalars []fr.Element) (*bn254.G1Affine, error) {
+	iciclePoints := HostSliceFromPoints(gnarkPoints)
+	icicleScalars := HostSliceFromScalars(gnarkScalars)
+
+	cfg := core.GetDefaultMSMConfig()
```

Comment:
We should expose this in the API so users can use streams and change the primitive config depending on their own needs (like setting that points and/or scalars are in Montgomery form).